apps/rush-lib/src/api/RushConfigurationProject.ts

```diff
@@ -227,25 +228,55 @@ export class RushConfigurationProject {
 
   /**
    * A list of projects within the Rush configuration which directly depend on this package.
+   * @deprecated Use downstreamDependencyProjectSet instead
    */
   public get downstreamDependencyProjects(): string[] {
+    return [...this._downstreamDependencyProjects];
+  }
+
+  /**
+   * A set of projects within the Rush configuration which directly depend on this package.
+   */
+  public get downstreamDependencyProjectSet(): Set<string> {
     return this._downstreamDependencyProjects;
   }
 
   /**
    * A map of projects within the Rush configuration which are directly depended on by this project
+   * @deprecated Use localDependencyProjectSet instead
    */
   public get localDependencyProjects(): ReadonlyArray<RushConfigurationProject> {
+    return [...this.localDependencyProjectSet];
+  }
+
+  /**
+   * The set of projects within the Rush configuration which are directly depended on by this project
+   */
+  public get localDependencyProjectSet(): ReadonlySet<RushConfigurationProject> {
     if (!this._localDependencyProjects) {
-      this._localDependencyProjects = [
-        ...this._getLocalDependencyProjects(this.packageJson.dependencies),
-        ...this._getLocalDependencyProjects(this.packageJson.devDependencies),
-        ...this._getLocalDependencyProjects(this.packageJson.optionalDependencies)
-      ];
+      const self: RushConfigurationProject = this;
+      this._localDependencyProjects = new Set(
+        (function* () {
+          yield* self._getLocalDependencyProjects(self.packageJson.dependencies);
+          yield* self._getLocalDependencyProjects(self.packageJson.devDependencies);
+          yield* self._getLocalDependencyProjects(self.packageJson.optionalDependencies);
+        })()
+      );
     }
     return this._localDependencyProjects;
   }
 
+  /**
+   * The set of projects withint he rush configuration which directly depend on this project.
+   * Excludes those that declare this project as a cyclicDependencyProject
+   */
+  public get localDependentProjectSet(): ReadonlySet<RushConfigurationProject> {
```

Comment:
Is this important to expose as a `@public` API?

I guess what I'm wondering is *"What would I use this API for?"*  If there's a meaningful use case, we could explain it in the `@remarks` section.

Reply:
I can switch to `@internal`. It's the companion of `localDependencyProjectSet`, which is part of the public API. The less useful form, `downstreamProjects` is already public, though maybe that should be revised?

Reply:
This is now `consumingProjects`, and left public so that consumers of rush-lib can walk the tree in both directions if desired.